app/src/pages/inside/stepPage/modals/editToInvestigateDefectModal/editToInvestigateDefectModal.jsx

```diff
@@ -0,0 +1,508 @@
+    btsIntegrations: PropTypes.array.isRequired,
+    data: PropTypes.shape({
+      item: PropTypes.array,
+      fetchFunc: PropTypes.func,
+    }).isRequired,
+    showNotification: PropTypes.func.isRequired,
+    hideModalAction: PropTypes.func.isRequired,
+    fetchTestItemsAction: PropTypes.func.isRequired,
+    unlinkIssueAction: PropTypes.func.isRequired,
+    linkIssueAction: PropTypes.func.isRequired,
+    postIssueAction: PropTypes.func.isRequired,
+    currentLaunch: PropTypes.object,
+    currentFilter: PropTypes.object,
+    tracking: PropTypes.shape({
+      trackEvent: PropTypes.func,
+      getTrackingData: PropTypes.func,
+    }).isRequired,
+  };
+
+  static defaultProps = {
+    currentLaunch: {},
+    currentFilter: null,
+  };
+
+  constructor(props) {
+    super(props);
+    const {
+      intl,
+      btsIntegrations,
+      data: { item },
+    } = props;
+    this.state = {
+      searchMode: SEARCH_MODES.CURRENT_LAUNCH,
+      selectedItems: [],
+      testItems: [],
+      changeCommentMode:
+        getStorageItem(CURRENT_CHANGE_DEFECT_COMMENT_MODE) || CHANGE_COMMENT_MODE.NOT_CHANGE,
+      ignoreAnalyzer: item.issue.ignoreAnalyzer,
+      markdownValue: item.issue.comment || '',
+      defectType: item.issue.issueType,
+      loading: false,
+    };
+
+    this.multiActionButtonItems = [
+      {
+        label: intl.formatMessage(messages.saveAndPostIssueMessage),
+        value: 'Post',
+        onClick: () => this.onEditDefects(this.handlePostIssue, true),
+        disabled: !isPostIssueActionAvailable(btsIntegrations),
+      },
+      {
+        label: intl.formatMessage(messages.saveAndLinkIssueMessage),
+        value: 'Link',
+        onClick: () => this.onEditDefects(this.handleLinkIssue, true),
+        disabled: !btsIntegrations.length,
+      },
+      {
+        label: intl.formatMessage(messages.saveAndUnlinkIssueMessage),
+        value: 'Unlink',
+        onClick: () => this.onEditDefects(this.handleUnlinkIssue, true),
+        disabled: !item.issue.externalSystemIssues || !item.issue.externalSystemIssues.length,
+      },
+    ];
+  }
+
+  componentDidMount() {
+    this.fetchSimilarDefects(this.state.searchMode);
+  }
+
+  onEditDefects = (nextAction, issueAction) => {
+    if (this.checkIfTheDataWasChanged()) {
+      this.props.tracking.trackEvent(STEP_PAGE_EVENTS.EDIT_DESCRIPTION_EDIT_DEFECT_MODAL);
+      this.saveDefects(this.prepareDataToSend());
+    }
+    !issueAction && this.props.tracking.trackEvent(STEP_PAGE_EVENTS.SAVE_BTN_EDIT_DEFECT_MODAL);
+    issueAction && this.props.hideModalAction();
+    nextAction();
+  };
+
+  getCloseConfirmationConfig = () => {
+    if (!this.checkIfTheDataWasChanged()) {
+      return null;
+    }
+    return {
+      confirmationWarning: this.props.intl.formatMessage(COMMON_LOCALE_KEYS.CLOSE_MODAL_WARNING),
+    };
+  };
+
+  getChangeCommentModeOptions = () => {
+    const { intl } = this.props;
+    return [
```

Comment:
Since this array is not changed in runtime, it can be a class field initialized in the constructor.